src/hotspot/share/opto/vectornode.cpp

```diff
@@ -1851,36 +1851,28 @@ Node* NegVNode::Ideal(PhaseGVN* phase, bool can_reshape) {
   return NULL;
 }
 
-Node* ReverseBytesVNode::Identity(PhaseGVN* phase) {
-  if (is_predicated_using_blend()) {
-    return this;
-  }
-  // ReverseBytesV (ReverseBytesV X , MASK) , MASK =>  X
-  if (in(1)->Opcode() == Op_ReverseBytesV) {
-    if (is_predicated_vector() && in(1)->is_predicated_vector() && in(2) == in(1)->in(2)) {
-      return in(1)->in(1);
-    } else {
-      // ReverseBytesV (ReverseBytesV X) =>  X
-      return in(1)->in(1);
+static Node* reverse_operations_identity(Node* n, Node* in1) {
+  if (n->is_predicated_using_blend()) {
+    return n;
+  }
+  if (n->Opcode() == in1->Opcode()) {
+    // OperationV (OperationV X , MASK) , MASK =>  X
+    if (n->is_predicated_vector() && in1->is_predicated_vector() && n->in(2) == in1->in(2)) {
+      return in1->in(1);
+    // OperationV (OperationV X) =>  X
+    } else if (!n->is_predicated_vector() && !in1->is_predicated_vector())  {
+      return in1->in(1);
```

Comment:
But this will still trigger the SonarCloud warning originally reported by @shipilev because both if and else branch contain the same code, right? Shouldn't the conditions be merged?

Reply:
I don't think it would trigger a warning. The original warning, as I understand it, was to say that the *unpredicated* `else` branch is the same. So we were guaranteed to take either of branches, and thus the same code, irrelevant of the predicate. It is not the same here: we now have a third path, going out without entering either branch :)

Reply:
Ah, right. That makes sense. I still think the branches could me merged but I don't have a strong opinion.

Reply:
Due to expression short circuiting and GCSE an optimizing compiler should produce same code with merged branches.